### DISCUSSION TITLE
Problem flashing BLE code

### DISCUSSION THREAD

**Discussion by fsalfner-dev** [rbaron/b-parasite#191] [Q&A] [answered]
Hi,

I have ordered a board from JLCPCB with all parts assembled and am using the J-Link EDU mini programmer.

I have connected the parasite according to this pinout plan:

![SWD pinout](https://cdn.sparkfun.com/assets/learn_tutorials/9/0/2/SWD_Header.png)

connecting the parasite's board VCC to pin 1, IO to pin 2, CLK to pin 4 and GND to pin 3 (according to the [Wiki](https://user-images.githubusercontent.com/1573409/206271100-475d544e-6a85-4a06-83f3-b325c6d0b3ed.png) ).

I was able to build the BLE sample code, but when I try to flash it I get the error

```
-- west flash: using runner nrfjprog
-- runners.nrfjprog: Flashing file: /home/felix/temp/b-parasite/code/nrf-connect/samples/ble/build/zephyr/zephyr.hex
[error] [ Client] - Encountered error -12: Command read_device_info executed for 13 milliseconds with result -12
[error] [ Client] - Encountered error -2: Command read_memory_descriptors executed for 0 milliseconds with result -2
Failed to read device memories.
[error] [ Worker] - Low voltage 0 detected in target device.
[error] [ Worker] - Cannot call read_access_port_register when connect_to_emu_without_snr or connect_to_emu_with_snr has not been called.
ERROR: Requested operation is currently unavailable.
ERROR: Check logs for more details.
NOTE: For additional output, try running again with logging enabled (--log).
NOTE: Any generated log error messages will be displayed.
FATAL ERROR: command exited with status 66: nrfjprog --program /home/felix/temp/b-parasite/code/nrf-connect/samples/ble/build/zephyr/zephyr.hex --sectoranduicrerase --verify -f NRF52 --snr 801052098
```

I presume the line saying `Low voltage 0 detected in target device` indicates the error.

When I power the parasite with the battery (and don't connect VCC), I receive a similar error, but with the message `Low voltage 1316 detected in target device`. Running it multiple times, the value `1316` varies a little (don't know if that is actually mV)

I first thought that there's a short circuit on the board, but with inserted battery the voltage between VCC and GND is stable around 3.1V.

Any idea what I am doing wrong or how I can figure out what the problem is?

**Comment by rbaron** [comment]
Hey @fsalfner-dev,

I believe the J-Link EDU Mini cannot power the target. The actual pinout is:

![JLink EDU Mini](https://embeddedcomputing.weebly.com/uploads/1/1/6/2/11624344/j-link-9-pin-cortex-m-adapter-1_1_orig.png)

The main difference being pin 1 -- VTref instead of VCC. So this explains why you're seeing 0 volts in the first scenario, as the target has no power.

For the second scenario, I would expect it to work. I would:
1. Measure the Vcc voltage when the J-Link is connected, ideally as "seen" by the J-Link. Maybe there's a short in the programmer connection, or maybe faulty solder joints/wire connections
2. Try it with a different b-parasite board, so you rule out components/soldering defects, specially if you soldered stuff by hand

Let us know how it goes.

**Comment by fsalfner-dev** [comment, marked as answer]
Thank you very much @rbaron .

Your hint that the J-Link Edu mini cannot power the parasite, and that the first pin is VTref rather than VCC helped me to find a solution:

I had to connect VCC to VTref *and* power the parasite with the battery.

Since the Wiki suggests to not connect both at the same time, I might add a hint to the wiki, too.

Thanks @rbaron, this is an awesome project!
